doc/tutorials/specification-language.md

```diff
@@ -0,0 +1,97 @@
+
+E.g. you can see
+that in the upper example we also mounted the [type](/src/plugins/type/README.md) plugin for validation checks. This for example
+can check for wrong types:
+
+```shell script
+kdb meta-set spec/tests/bool 'type' boolean
+kdb set /tests/bool Batman
+# Sorry, module type issued the error C03200:
+# Validation Semantic: The value 'Batman' of key 'user/tests/bool' could not be converted into a boolean
+```
+
+Please note as of Elektra v0.9 you cannot add an unlimited amount of plugins. You can see if this issue
+still remains in our issue tracker([#2133](https://github.com/ElektraInitiative/libelektra/issues/2133)).
+
+## Specification writing
+
+There are multiple ways how you can set constraints for configuration settings. All possibilities though set `metadata` to certain
+configuration settings (this is also done in the spec namespace). A metadata is just additional data describing the configuration setting.
+One way to set metadata is the command line tool [kdb meta-set](../help/kdb-meta-set.md) which we have used in the upper example.
+For the [validation plugin](../../src/plugins/validation/README.md) which supports regular expression checks there exists even its own
+command line tool [kdb vset](../help/kdb-vset.md).
+
+If you are writing validation for dozens of configuration settings, the command line might not be the most convenient way. You can also
+write specifications in any file format of your desire. Examples can be seen [here](../../examples/spec). In Elektra, all file formats
+can have metadata but some needed extra tinkering like any `yaml` parser since metadata are not natively supported. We recommend the `ini`
+file format as it natively supports metadata. So the upper example
+could be rewritten in an `ini` file in the following way:
+
+```ini
+[]
+mountpoint = test-specification.ni
+infos/plugins = type
+
+[bool]
+type = boolean
+```
+
+To integrate a configuration into the key database, we mount it by using the property `mountpoint` in the configuration specification.
+This will be mounted into the spec namespace with the following command:
+
+```shell script
+kdb mount </absolute/path/to/ini/file.ini> spec/tests ni
+kdb spec-mount /tests
+```
+
+The `spec-mount` command tells Elektra to load all relevant plugins. It is smart enough to automatically detect the needed plugins
+by looking at all metadata. You might need to remove some metadata to not load certain plugins that cause the exceeding of the maximum
+plugin number.
+
+What remains to know are the metadata names and meanings. We have used the `type` plugin which provides the metadata `type` metadata along with
+many other metadata such as `check/enum`. You can read all available metadata in the plugin READMEs which all have to have a
+so called `contract` at the beginning of each file. This contract also contains the `infos/metadata` data which lists all available
```

Comment:
Continue here with the example of the type plugin. Also give another example, I do not think this text makes the concept "contract" clear.